src/components/map/map-legend.js

```diff
@@ -67,14 +69,23 @@ export const StyledMapControlLegend = styled.div`
 export const VisualChannelMetric = ({name}) => (
   <div className="legend--layer__title">
     <span className="legend--layer_by">by </span>
-    <span className="legend--layer_color_field">{name}</span>
+    <span className="legend--layer_color_field">
+      <FormattedHTMLMessage
+        id={name}
+        values={{
+          name: useIntl().formatMessage({id: `property.${camelize(name)}`, defaultMessage: name})
```

Comment:
whenever in function stateless components you should always try to take advantage of useMemo and useCallback.
in this case you can do the following:
```
const values = useMemo(() => ({
  name: useIntl().formatMessage({id: `property.${camelize(name)}`, defaultMessage: name})
}), [name]);
```